core/src/main/scala/reconciler/Reconciler.scala

```diff
@@ -119,6 +119,88 @@ abstract class Reconciler[S <: CustomResource[_, _], T <: ObjectResource] {
   }
   // scalastyle:on
 
+  case class CacheEntry(s: Option[Source], ts: List[Target])
```

Comment:
be nice to be able to support multiple targets... but maybe that is some we can do later.

Reply:
Agreed. That's one of the things I'm trying to implement using a shapeless Coproduct based flow.